### DISCUSSION TITLE
Disk cache dimension with 150k static pages

### DISCUSSION THREAD

**Discussion by FrancescoBonizzi** [vercel/next.js#62085] [Help] [answered]
### Summary

We are currently hosting a NextJs website on a Linux Azure WebApp. It's a recipes website ([if you need to look it, it's here](https://ricette.tuduu.it)). We aren't using App Router. 

Our biggest route is `/pages/[slug]/index.tsx`, here we generate at build time 200 pages, then we let the server generate other routes content on demand, then cache them (`fallback: blocking`). The problem is that with so many pages (more than 150k) after two weeks the WebApp storage is filled with more than 20Gb of cached pages - I navigated the WebApp, everything is in `.next/server/pages` folder. With this kind of size, our deploy task (`AzureRmWebAppDeployment@4`) hangs a lot and usually timeouts (I really don't know way, but empirically that's the reason), so we need to fill less space. 

The ideal solution could be to conditionally cache on disk some pages, for example I would keep in cache the build time generated pages and a bunch of other pages (for example all the recipes created in the last month only, since we have creators that are always creating new contents). But it doesn't seem possible.

Do you have any other ideas? Is there something we could do better?

Thanks.

### Additional information

```js
(Development machine)

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:34 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8103
Binaries:
  Node: 21.1.0
  npm: 10.2.0
  Yarn: 1.22.19
  pnpm: 8.15.2
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by Franetse7** [comment]
Ok thanks

**Comment by FrancescoBonizzi** [comment]
🧑‍💻💻🌃😔📅

**Comment by SouthLink** [comment, marked as answer]
There are several caching strategies you can use to control the number of pages you have:

First of all, [configuration-caching](https://nextjs.org/docs/pages/building-your-application/deploying#configuring-caching) for next.js should give you custom control over your web cache, if you need more fine-grained control you may want to [custom-server](https://nextjs.org/docs/pages/building-your-application/configuring/custom-server)

Strategy 1:
You can control the number of pages you have by using the [Least Recently Used (LRU) cache](https://en.wikipedia.org/wiki/Cache_replacement_policies#LRU), which is more accurate.

Strategy 2:
You can control your cache by setting the expiration time of your web cache,  which may be used in conjunction with other databases, such as redis， and you may have to adjust it several times to get the best configuration

example:

```
class RedisCache {
  constructor(client) {
    this.client = client;
  }
  // ...
  set(key, val, expireMode = "EX", expireTime = 600) {
    return new Promise((resolve, reject) => {
      this.client.set(key, val, expireMode, expireTime, (err, res) => {
        if (err) reject(err);
        resolve(res);
      });
    });
  }

}

module.exports = RedisCache;

```
